my_enums.rb

```diff
@@ -1,60 +1,183 @@
 # frozen_string_literal: true
 
-# rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+# rubocop: disable Metrics/AbcSize
+# rubocop: disable Metrics/CyclomaticComplexity
+# rubocop: disable Metrics/MethodLength
+# rubocop: disable Metrics/PerceivedComplexity
 
 module Enumerable #:nodoc: all
   def my_each
-    length.times do |n|
-      current = self[n]
+    return to_enum unless block_given?
+
+    arr = to_a if is_a? Range
+    arr = self unless is_a? Range
+    arr.length.times do |n|
+      current = arr[n]
       yield(current)
     end
+    self
   end
 
   def my_each_with_index
-    length.times do |n|
-      index = n
-      current = self[n]
-      yield(index, current)
+    return to_enum unless block_given?
+
+    my_each do |n|
+      yield(n, index(n))
     end
+    self
   end
 
   def my_select
     emp_arr = []
+    return to_enum unless block_given?
+
     my_each do |n|
       emp_arr << n if yield(n)
     end
     emp_arr
   end
 
-  def my_all?
-    emp_arr = []
-    my_each do |n|
-      emp_arr << n if yield(n)
+  def my_all?(param = nil)
+    if block_given?
+      begin
+        if yield.is_a? Class
+          my_each do |n|
+            return false unless n.is_a? yield
+          end
+          true
+        elsif yield.instance_of? Regexp
+          my_each do |n|
+            next if n =~ yield
+
+            return false
+          end
+          true
+        end
+      rescue NoMethodError
+        my_each do |n|
+          next if yield(n)
+
+          return false
+        end
+        true
+      end
+    elsif !param.nil?
+      if param.is_a? Class
+        my_each do |n|
+          return false unless n.is_a? param
+        end
+      elsif param.instance_of? Regexp
+        my_each do |n|
+          next if n =~ param
+
+          return false
+        end
+      else
+        my_each do |n|
+          next if n == param
+
+          return false
+        end
+      end
+      true
+    else
+      return false if empty?
+
+      my_each do |n|
+        return false if n.nil? || n == false
+      end
+      true
     end
-    emp_arr == self
   end
 
-  def my_any?
-    tester = false
-    my_each do |n|
-      tester = true if yield(n)
-      break if tester
+  def my_any?(param = nil)
+    if block_given?
+      begin
+        if (yield.is_a? Class) || (yield.instance_of? Regexp)
+          my_each do |n|
+            return true if (n.is_a? yield) || (n =~ yield)
+          end
+          false
+        end
+      rescue NoMethodError
+        my_each do |n|
+          return true if yield(n)
+        end
+        false
+      end
+    elsif !param.nil?
+      my_each do |n|
+        return true if (n.class == param) || (n =~ param) || (n == param)
+      end
+      false
+    else
+      return false if empty?
+
+      my_each do |n|
+        return true unless n.nil? || n == false
+      end
+      false
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method doesn't consider the possibility when a string is given as an argument `(some_enumerable.my_any?('cow')).`